main.go

```diff
@@ -127,15 +127,14 @@ func checkBitswapCID(ctx context.Context, c cid.Cid, ma multiaddr.Multiaddr) (*b
 	}
 
 	// in case for some reason we're sent a bunch of messages (e.g. wants) from a peer without them responding to our query
+	// FIXME: Why would this be the case?
```

Comment:
i.e. why would it take us 10 seconds to get a response to our query?

Reply:
btw if you're looking at this for debugging you might want to add a verbose option to the `bitswap check` command that prints out `res.msg.Loggable()` that will let you see if and what types of messages are coming in even if they aren't the responses we're looking for.

Reply:
the arbitrary 10 seconds comes from your original 3 loops with 3 seconds delay